ethereum/spec/src/main/java/tech/pegasys/teku/spec/Spec.java

```diff
@@ -69,12 +72,28 @@
 public class Spec {
   private final Map<SpecMilestone, SpecVersion> specVersions;
   private final ForkSchedule forkSchedule;
+  private final StateTransition stateTransition;
 
   private Spec(Map<SpecMilestone, SpecVersion> specVersions, final ForkSchedule forkSchedule) {
     Preconditions.checkArgument(specVersions != null && specVersions.size() > 0);
     Preconditions.checkArgument(forkSchedule != null);
     this.specVersions = specVersions;
     this.forkSchedule = forkSchedule;
+
+    // Setup state transition
+    final Map<UInt64, StateUpgrade<?>> slotToStateUpgrade = new HashMap<>();
+    forkSchedule
+        .streamMilestoneBoundarySlots()
+        .forEach(
+            milestoneAndSlot -> {
+              final SpecMilestone milestone = milestoneAndSlot.getLeft();
+              final UInt64 slot = milestoneAndSlot.getRight();
+              final SpecVersion specVersion = specVersions.get(milestone);
+              StateUpgrade.create(specVersion)
+                  .ifPresent(upgrade -> slotToStateUpgrade.put(slot, upgrade));
```

Comment:
Any reason we don't structure this as `specVersion.getStateUpgrade()`?

Reply:
No - that makes way more sense 😂